src/pages/api/pdf.ts

```diff
@@ -0,0 +1,101 @@
+import chromium from 'chrome-aws-lambda'
+import { NextApiRequest, NextApiResponse } from 'next'
+
+// this runs on chrome-aws-lambda and does not work on normal machines
+// run `yarn add -D puppeteer` to test locally,
+// but since it uses a lot of space make sure to remove it before merging
+
+const styles = `
+    width: 100%;
+    position:relative;
+    top: -10px;
+    font-size: 9px;
+    font-family: Arial, Helvetica, sans-serif;
+    display: flex;
+    justify-content: space-between;
+    padding-right:40px;
+    padding-left:40px;
+    color: #666;
+    `
+
+export default async function createPdf(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const noSolutions = req.query.noSolutions !== undefined
+  const urlString = decodeURIComponent(req.query.url.toString())
```

Comment:
Passing the full url to the route feels a bit awkward. Can we refactor this and only pass in the ID of the content?

You can get the absolute url of the deployment with a package like https://github.com/jakeburden/next-absolute-url

(for the language, we can probably also pass it in as query parameter ...)

Reply:
We could do it this way. But tbh. I'm not sure I share your awkward feeling.
We use the same style for fetching thumbnails:
https://embed.serlo.org/thumbnail?url=https%3A%2F%2Fwww.youtube-nocookie.com%2Fembed%2FKtV2wlp9Ts4%3Fautoplay%3D1%26html5%3D1
Also all the share links work the same way.

But I can be convinced :) 
Why do you preferre to use ids?

Reply:
okay, it now uses an id parameter and the absolute id.

Reply:
(also: it's not working any more -_-)
please don't merge

Reply:
This is working: https://frontend-git-pdf-print-test-serlo.vercel.app/api/pdf?id=1565

Reply:
super weird. 
this is not: https://frontend-git-pdf-print-test-serlo.vercel.app/api/pdf?id=1555
but it seems to be related to this page.
… okay, need to test more entities again. so weird because I'm sure this one worked without problems before.

Reply:
But revisions are not working: https://frontend-git-pdf-print-test-serlo.vercel.app/api/pdf?id=212696

Reply:
thats fine. we don't offer that.

Reply:
still not sure what happens on 1555…

Reply:
okay, just to document here:
1555 results in a >22MB PDF locally. Which is too big for AWS (and too big in general).
Probably something with the images, maybe they convert it from webp to something else and don't mind the size?

Reply:
```
[ERROR] [1628665754926] LAMBDA_RUNTIME Failed to post handler success response. Http response code: 413.
[ERROR] [1628665754951] LAMBDA_RUNTIME Failed to get next invocation. Http Response code: 403
Unexpected Top Level Error: Error: Failed to get next invocation, error 403
[ERROR] [1628665754963] LAMBDA_RUNTIME Failed to post handler success response. Http response code: 403.
RequestId: 3a2fb12c-2a39-4d24-a2f5-493e5ba16099 Error: Runtime exited with error: exit status 129
Runtime.ExitError
```

Reply:
interesting! The pdf-size of 1565 is <1MB and looks fine ... and the images for 1555 aren't that big, either.